sympy/core/mul.py

```diff
@@ -1001,13 +1003,15 @@ def _matches_expand_pows(arg_list):
         return new_args
 
     @staticmethod
-    def _matches_noncomm(nodes, targets, repl_dict={}):
+    def _matches_noncomm(nodes, targets, repl_dict=None):
         """Non-commutative multiplication matcher.
 
         `nodes` is a list of symbols within the matcher multiplication
         expression, while `targets` is a list of arguments in the
         multiplication expression being matched against.
         """
+        if repl_dict is None:
+            repl_dict = {}
         repl_dict = repl_dict.copy()
```

Comment:
Note that the change here is completely redudant, as the copy on the next line ensures that no mutation can occur